v2/bigquery-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/BigQueryToSpanner.java

```diff
@@ -0,0 +1,232 @@
+ * APP_ROOT=/path/to/app-root
+ * COMMAND_SPEC=/path/to/command-spec
+ *
+ * # Build and upload image
+ * mvn clean package \
+ * -Dimage=${TARGET_GCR_IMAGE} \
+ * -Dbase-container-image=${BASE_CONTAINER_IMAGE} \
+ * -Dbase-container-image.version=${BASE_CONTAINER_IMAGE_VERSION} \
+ * -Dapp-root=${APP_ROOT} \
+ * -Dcommand-spec=${COMMAND_SPEC}
+ *
+ * # Create an image spec in GCS that contains the path to the image
+ * {
+ *    "docker_template_spec": {
+ *       "docker_image": $TARGET_GCR_IMAGE
+ *     }
+ *  }
+ *
+ * # Execute template:
+ * API_ROOT_URL="https://dataflow.googleapis.com"
+ * TEMPLATES_LAUNCH_API="${API_ROOT_URL}/v1b3/projects/${PROJECT}/templates:launch"
+ * JOB_NAME="bigquery-to-spanner-`date +%Y%m%d-%H%M%S-%N`"
+ * time curl -X POST -H "Content-Type: application/json"     \
+ *      -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+ *      "${TEMPLATES_LAUNCH_API}"`
+ *      `"?validateOnly=false"`
+ *      `"&dynamicTemplate.gcsPath=gs://path/to/image/spec"`
+ *      `"&dynamicTemplate.stagingLocation=gs://path/to/stagingLocation" \
+ *      -d '
+ *       {
+ *         "jobName":"'$JOB_NAME'",
+ *         "parameters": {
+ *           "bigQueryTables" "'$BIG_QUERY_TABLES'",
+ *           "spannerTables": "$SPANNER_TABLES",
+ *           "spannerDatabase": "$SPANNER_DATABASE",
+ *           "spannerInstance": "$SPANNER_INSTANCE"
+ *         }
+ *       }
+ *       '
+ * </pre>
+ */
+public class BigQueryToSpanner {
+
+  /* Logger for class. */
+  private static final Logger LOG = LoggerFactory.getLogger(BigQueryToSpanner.class);
+
+  /**
+   * File suffix for file to be written.
+   */
+  private static final String FILE_SUFFIX = ".parquet";
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    BigQueryToSpannerOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(BigQueryToSpannerOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs the pipeline with the supplied options.
+   *
+   * @param options The execution parameters to the pipeline.
+   * @return The result of the pipeline execution.
+   */
+  private static PipelineResult run(BigQueryToSpannerOptions options) {
+
+    // Create the pipeline.
+    Pipeline pipeline = Pipeline.create(options);
+
+    List<String> bigQueryTables = options.getBigQueryTables();
+    List<String> spannerTables = options.getSpannerTables();
+    if (spannerTables == null) {
+      spannerTables = new ArrayList<>();
+      for (String ref : bigQueryTables) {
+        spannerTables.add(BigQueryHelpers.parseTableSpec(ref).getTableId());
+      }
+    } else {
+      Preconditions.checkArgument(bigQueryTables.size() == spannerTables.size(),
+          "spannerTables option must have the same number of tables as bigQueryTableRefs if provided");
+    }
+
+    for (int i = 0; i < bigQueryTables.size(); i++) {
```

Comment:
there is no sentence that tables that are migrated from,to should be in same order :)